File: app/src/main/java/com/dublikunt/nclient/async/downloader/DownloadObserver.java
package com.dublikunt.nclient.async.downloader;

public interface DownloadObserver {
    void triggerStartDownload(GalleryDownloader downloader);

    void triggerUpdateProgress(GalleryDownloader downloader, int reach, int total);

    void triggerEndDownload(GalleryDownloader downloader);

    void triggerCancelDownload(GalleryDownloader downloader);

    void triggerPauseDownload(GalleryDownloader downloader);
}

File: app/src/main/kotlin/com/dublikunt/nclient/async/downloader/DownloadObserver.kt
package com.dublikunt.nclient.async.downloader

interface DownloadObserver {
    fun triggerStartDownload(downloader: GalleryDownloader)
    fun triggerUpdateProgress(downloader: GalleryDownloader?, reach: Int, total: Int)
    fun triggerEndDownload(downloader: GalleryDownloader?)
    fun triggerCancelDownload(downloader: GalleryDownloader)
    fun triggerPauseDownload(downloader: GalleryDownloader?)
}
